OpenRA.Mods.Common/ServerTraits/MasterServerPinger.cs

```diff
@@ -23,35 +24,52 @@ public class MasterServerPinger : ServerTrait, ITick, INotifySyncLobbyInfo, ISta
 	{
 		// 3 minutes. Server has a 5 minute TTL for games, so give ourselves a bit of leeway.
 		const int MasterPingInterval = 60 * 3;
+		static readonly Beacon LanGameBeacon = new Beacon("OpenRALANGame", (ushort)new Random(DateTime.Now.Millisecond).Next(2048, 60000));
+
 		public int TickTimeout { get { return MasterPingInterval * 10000; } }
 
+		long lastPing = 0;
+		bool isInitialPing = true;
+		bool firstTick = true;
+
+		volatile bool isBusy;
+		Queue<string> masterServerMessages = new Queue<string>();
+
 		public void Tick(S server)
 		{
+			if (firstTick)
```

Comment:
You can implement `INotifyServerStart` and then put this inside `INotifyServerStart.ServerStarted(Server s)`.